src/server/api/index.js

```diff
@@ -268,19 +352,66 @@ function createAPI(app, mountPath, middlewareOpts) {
             userId: req.params.username,
             email: req.body.email,
             password: req.body.password,
-            canCreate: req.body.canCreate || false
+            canCreate: req.body.canCreate || false,
+            data: req.body.data || {}
         };
 
         putUser(receivedData, req, res, next);
     });
 
     router.patch('/users/:username', function (req, res, next) {
+        // body params
+        //"email": "user@example.com",
+        //"password": "pass",
+        //"canCreate": null,
+        //"siteAdmin": false,
+        //"data": {}
+        ensureSameUserOrSiteAdmin(req, res)
+            .then(function (userData) {
+                if (req.body.hasOwnProperty('siteAdmin') && !userData.siteAdmin) {
```

Comment:
I know that the old implementation had `!userData.siteAdmin` in the future we may need to use `userData.siteAdmin !== true` to be on the safe side.